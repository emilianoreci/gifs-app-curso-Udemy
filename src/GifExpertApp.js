//  @Package
import React, { useState } from "react";

//  @Own
import { AddCategories } from "./componets/AddCategories";
import { GifGrid } from "./componets/GifGrid";

const GifExpertApp = ({ defaultCategory = [] }) => {
  //const [categories, setCategories] = useState(["goku"]);
  //En vez de poner directamente goku en el useState, lo paso por propsDefault
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategories setCategories={setCategories} />

      {categories.map((each) => (
        <GifGrid key={each} category={each} />
      ))}
    </>
  );
};

export default GifExpertApp;
