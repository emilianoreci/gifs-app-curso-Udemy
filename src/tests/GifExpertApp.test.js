import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas al archivo GifExpertApp", () => {
  test("Debe mostrar el componente correctamente", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrarse una lista de categorias", () => {
    const categories = ["batman", "sun"];

    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);

    expect(wrapper).toMatchSnapshot();

    //espera que encuentre tantos GifGrid como categories.length
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
