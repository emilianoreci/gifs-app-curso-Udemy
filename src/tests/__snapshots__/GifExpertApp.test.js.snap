// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al archivo GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategories
    setCategories={[Function]}
  />
</Fragment>
`;

exports[`Pruebas al archivo GifExpertApp debe de mostrarse una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategories
    setCategories={[Function]}
  />
  <GifGrid
    category="batman"
    key="batman"
  />
  <GifGrid
    category="sun"
    key="sun"
  />
</Fragment>
`;
